src/Views/DriversView.vala

```diff
@@ -0,0 +1,114 @@
+            hexpand = true,
+            max_width_chars = 1, // Make Gtk wrap, but not expand the window
+            wrap = true,
+            xalign = 0
+        };
+
+        var legal_image = new Gtk.Image.from_icon_name ("text-x-copying-symbolic", LARGE_TOOLBAR);
+
+        var legal_label = new Gtk.Label (_("Proprietary drivers are subject to their own license terms. By installing, you agree to their terms of use.")) {
+            hexpand = true,
+            max_width_chars = 1, // Make Gtk wrap, but not expand the window
+            wrap = true,
+            xalign = 0
+        };
+
+        var install_later_image = new Gtk.Image.from_icon_name ("system-software-install-symbolic", LARGE_TOOLBAR);
+
+        var install_later_label = new Gtk.Label (_("Proprietary drivers can be installed later through AppCenter, but this will require a working Internet connection.")) {
+            hexpand = true,
+            max_width_chars = 1, // Make Gtk wrap, but not expand the window
+            wrap = true,
+            xalign = 0
+        };
+
+        var drivers_label = new Gtk.Label (_("Install Proprietary Drivers:"));
+
+        var drivers_switch = new Gtk.Switch ();
+
+        var drivers_box = new Gtk.Box (HORIZONTAL, 12) {
+            halign = CENTER
+        };
+        drivers_box.add (drivers_label);
+        drivers_box.add (drivers_switch);
+
+        var message_grid = new Gtk.Grid () {
+            valign = CENTER,
+            row_spacing = 32,
+            column_spacing = 12
+        };
+        message_grid.attach (description_label, 0, 0, 2, 1);
+        message_grid.attach (warning_image, 0, 1);
+        message_grid.attach (warning_label, 1, 1);
+        message_grid.attach (legal_image, 0, 2);
+        message_grid.attach (legal_label, 1, 2);
+        message_grid.attach (install_later_image, 0, 3);
+        message_grid.attach (install_later_label, 1, 3);
+        message_grid.attach (drivers_box, 0, 4, 2, 1);
+
+        content_area.column_homogeneous = true;
+        content_area.margin_start = content_area.margin_end = 12;
+        content_area.valign = Gtk.Align.CENTER;
+        content_area.attach (image, 0, 0);
+        content_area.attach (title_label, 0, 1);
+        content_area.attach (message_grid, 1, 0, 1, 2);
+
+        var back_button = new Gtk.Button.with_label (_("Back"));
+        back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
+
+        var next_button = new Gtk.Button.with_label (_("Erase and Install"));
+        next_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        next_button.clicked.connect (() => next_step ());
+
+        action_area.add (back_button);
+        action_area.add (next_button);
+
+        drivers_switch.notify["active"].connect (() => {
+            unowned var configuration = Configuration.get_default ();
+            configuration.install_drivers = drivers_switch.active;
+        });
```

Comment:
I think since this is something we're including as part of a list (of things to install later) and not a feature that we're toggling on and off in real time, a checkbox is probably the more appropriate widget. Also, Ubiquity seems to hammer in the third-party-ness of the drivers here, which seems like a good idea:

![Screenshot from 2023-07-26 09 47 22](https://github.com/elementary/installer/assets/7277719/09e7445c-2260-47c6-b6c8-7cb14a942df7)

```suggestion
        var drivers_check = new Gtk.CheckButton.with_label (_("Include third-party proprietary drivers when installing"));

        var message_grid = new Gtk.Grid () {
            valign = CENTER,
            row_spacing = 32,
            column_spacing = 12
        };
        message_grid.attach (description_label, 0, 0, 2, 1);
        message_grid.attach (warning_image, 0, 1);
        message_grid.attach (warning_label, 1, 1);
        message_grid.attach (legal_image, 0, 2);
        message_grid.attach (legal_label, 1, 2);
        message_grid.attach (install_later_image, 0, 3);
        message_grid.attach (install_later_label, 1, 3);
        message_grid.attach (drivers_check, 0, 4, 2, 1);

        content_area.column_homogeneous = true;
        content_area.margin_start = content_area.margin_end = 12;
        content_area.valign = Gtk.Align.CENTER;
        content_area.attach (image, 0, 0);
        content_area.attach (title_label, 0, 1);
        content_area.attach (message_grid, 1, 0, 1, 2);

        var back_button = new Gtk.Button.with_label (_("Back"));
        back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);

        var next_button = new Gtk.Button.with_label (_("Erase and Install"));
        next_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
        next_button.clicked.connect (() => next_step ());

        action_area.add (back_button);
        action_area.add (next_button);

        drivers_check.toggled.connect (() => {
            unowned var configuration = Configuration.get_default ();
            configuration.install_drivers = drivers_check.active;
        });
```